edb/edgeql/compiler/expr.py

```diff
@@ -549,10 +552,50 @@ def compile_UnaryOp(
 def compile_GlobalExpr(
     expr: qlast.GlobalExpr, *, ctx: context.ContextLevel
 ) -> irast.Set:
-    glob = ctx.env.get_schema_object_and_track(
-        s_utils.ast_ref_to_name(expr.name), expr.name,
-        modaliases=ctx.modaliases, type=s_globals.Global)
-    assert isinstance(glob, s_globals.Global)
+    # The expr object can be either a Permission or Global.
+    # Get an Object and manually check for correct type and None.
+    expr_schema_name = s_utils.ast_ref_to_name(expr.name)
+    expr_obj = ctx.env.get_schema_object_and_track(
+        expr_schema_name,
+        expr.name,
+        default=None,
+        modaliases=ctx.modaliases,
+        type=so.Object,
+    )
+
+    # Check None
+    if expr_obj is None:
+        s_schema.Schema.raise_bad_reference(
+            expr_schema_name,
+            module_aliases=ctx.modaliases,
+            span=expr.span,
+            type=s_globals.Global,
+        )
```

Comment:
Document why you are doing this instead of letting `get` handle it. (I think to get `global` mentioned in the error`?